File: code_samples/lua/utilities_regex.lua
-- Utilities: Regex
-- AI/ML Training Sample

Regex = {}
Regex.__ index = Regex

function Regex:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Regex:process(input)
    self.data = input
end

function Regex:getData()
    return self.data
end

function Regex:validate()
    return #self.data > 0
end

-- Example usage
local instance = Regex:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/elixir/utilities_regex.ex
# Utilities: Regex
# AI/ML Training Sample

defmodule Regex do
  defstruct data: ""
  
  def new(), do: %Regex{}
  
  def process(%Regex{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Regex{data: data}), do: data
  
  def validate(%Regex{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Regex.new()
updated = Regex.process(instance, "example")
IO.puts("Data: " <> Regex.get_data(updated))
IO.puts("Valid: " <> to_string(Regex.validate(updated)))
